### PR TITLE
Send Emails Asynchronously

### PR DESCRIPTION
**Pull request by dvaldivia**

### CODE REVIEW

File: cluster/scheduler.go [hunk 1]
```diff
@@ -127,28 +133,32 @@ func markTask(task *Task, newStatus TaskStatus) error {
 		`UPDATE tasks 
 					SET status = $1 %s
```
Comment (Alevsk) [inline cluster/scheduler.go:134]: what is the `%s` ? can we parametrize that?

**Review by Alevsk — APPROVED** [review]

File: cluster/email.go [hunk 2]
```diff
@@ -232,3 +236,50 @@ func SetEmailTemplate(ctx *Context, templateName, templateBody string) error {
...
+		return err
+	}
+	log.Println("asdasd")
+	// perform invitation
+	err = doInviteUserByEmail(ctx, taskData.UserFor, &user)
+	if err != nil {
+		log.Println("3")
+		log.Println(err)
+		if err := ctx.Rollback(); err != nil {
+			log.Println("4")
```
Comment (cesnietor) [inline cluster/email.go:279]:
```suggestion
```

File: cluster/email.go [hunk 3]
```diff
@@ -232,3 +236,50 @@ func SetEmailTemplate(ctx *Context, templateName, templateBody string) error {
...
+	UserID   string
+}
+
+// InviteUserByEmailTask performs the invitation of a tenant by email
+func InviteUserByEmailTask(task *Task) error {
+	// hydrate the data from the task
+	var taskData InviteUserTaskData
+	err := json.Unmarshal(task.Data, &taskData)
+	if err != nil {
+		log.Println("2")
```
Comment (Alevsk) [inline cluster/email.go:252]: I think this should be changed to `log.Println(err)`

File: cluster/email.go [hunk 4]
```diff
@@ -232,3 +236,50 @@ func SetEmailTemplate(ctx *Context, templateName, templateBody string) error {
...
+		return err
+	}
+	tenantID, _ := uuid.FromString(taskData.TenantID)
+	tenant, err := GetTenantByID(&tenantID)
+	if err != nil {
+		log.Println("2")
```
Comment (Alevsk) [inline cluster/email.go:258]: I think this should be changed to `log.Println(err)`

File: cluster/email.go [hunk 5]
```diff
@@ -232,3 +236,50 @@ func SetEmailTemplate(ctx *Context, templateName, templateBody string) error {
...
+		return err
+	}
+	ctx := NewCtxWithTenant(&tenant)
+	userID, err := uuid.FromString(taskData.UserID)
+	if err != nil {
+		log.Println("2")
```
Comment (Alevsk) [inline cluster/email.go:264]: I think this should be changed to `log.Println(err)`

File: cluster/admins.go [hunk 6]
```diff
@@ -64,28 +66,49 @@ func AddAdminAction(ctx *Context, name string, adminEmail string) (*Admin, error
...
 	}
 
+	// send an email to the admin
+	if err := SendAdminInvite(ctx, &admin, adminToken); err != nil {
+		return nil, err
+	}
+	return &admin, nil
+}
+
+func SendAdminInvite(ctx *Context, admin *Admin, adminToken *uuid.UUID) error {
```
Comment (cesnietor) [inline cluster/admins.go:76]: can you add comments to the functions

File: cluster/email.go [hunk 7]
```diff
@@ -232,3 +236,50 @@ func SetEmailTemplate(ctx *Context, templateName, templateBody string) error {
...
+		return err
+	}
+	user, err := GetUserByID(ctx, userID)
+	if err != nil {
+		log.Println("1")
```
Comment (Alevsk) [inline cluster/email.go:269]: I think this should be changed to `log.Println(err)`

File: cluster/email.go (hunk 2 above)
Comment (Alevsk) [inline cluster/email.go:279]: I think this should be changed to `log.Println(err)`

**Review by Alevsk — CHANGES_REQUESTED** [review]

**Pull request merged**